loop/tests/test_backtest.py

```
@@ -0,0 +1,216 @@
+        mock_app = mocker.MagicMock()
+        mock_dash.return_value = mock_app
+        
+        # Create mock predictor and features
+        mock_predictor = MockPredictor()
+        mock_features = MockFeatures()
+        
+        # Create backtester and run
+        backtest = BackTest(mock_predictor, mock_features)
+        
+        # Mock results for backtest
+        backtest.results = create_mock_results()
```

Ruff (0.8.2)

Undefined name `create_mock_results`

(F821)